\documentclass[a4paper]{article}
\usepackage{graphicx,amssymb}
\usepackage{authblk}

\textwidth=15cm \hoffset=-1.2cm
\textheight=25cm \voffset=-2cm

\pagestyle{empty}

\date{}

\def\keywords#1{\begin{center}{\bf Keywords}\\{#1}\end{center}}

\def\titulo#1{\title{#1}}
\def\autores#1{\author{#1}}

\begin{document}

\titulo{On Genomic Repeats and Reproducibility}

\author[1]{Can Firtina}
\author[1]{Can Alkan}
\affil[1]{\small Dept. of Computer Engineering, Bilkent University, 06800 Ankara, Turkey}
%\renewcommand\Authands{ and }

\maketitle
\thispagestyle{empty}

% The abstract

\begin{abstract}
The advancements in high throughput sequencing (HTS) technologies have increased the demand on producing  genome sequence data for many research questions. However, analyzing and interpreting HTS data with high accuracy is still a challenge. 
Although many algorithms were developed for this purpose, a handful of computational pipelines from mapping to variant calling may be considered standard, as they are commonly used in large-scale genome projects.

Approximately half of the human genome consists of repeats, which cause ambiguity in read mapping when the read length is short. On the average, a 100-bp read generated by the Illumina platform may align to hundreds of genome locations with similar edit distance. 
The BWA~\cite{Li2009a} mapper's  approach to handle such ambiguity is randomly selecting one location, and assigning the mapping quality to zero to inform the variant calling algorithms that the alignment may not be accurate. 

In this study, we investigated whether some of the commonly used variant discovery algorithms
make use of this mapping quality information, and how they react to genomic repeats.
We aligned two whole genome shotgun (WGS) datasets with one low (5X) and one high (45X) coverage genome
sequenced as part of the 1000 Genomes Project~\cite{1000GP2012} to the human reference genome (GRCh37) {\bf twice} using BWA with the same parameters. However, we shuffled the order of reads in the second mapping experiment to make sure that the same random number is not used for the same reads.
As expected, BWA reported different map locations to repetitive regions ($\sim$2.8\% of reads). Interestingly, although BWA reported zero mapping qualities for 
most of the discrepant read mappings ($\sim$94\%), it also assigned high MAPQ values ($\geq$30) for a fraction of them ($\sim$1.75\%). 
We then generated two single nucleotide polymorphism (SNP) and indel callsets each using GATK~\cite{DePristo2011} HaplotypeCaller, GATK UnifiedGenotyper, Freebayes~\cite{Garrison2012}, Platypus~\cite{Rimmer2014}, and SAMTools~\cite{Li2009b}, and structural variation (SV)
callsets using DELLY~\cite{Rausch2012} and LUMPY~\cite{Layer2014} from each genome.

We observed substantial differences in the callsets generated by all of the tools we tested.


%%%%%%%%%%%%% will continue


Our results show that all the variant callers but GATK HaplotypeCaller and GATK UnifiedGenotyper produce same variation discovery when the same individual's sequncing data is given as input. However, GATK HaplotypeCaller and GATK UnifiedGenotyper fail to be deterministic when the same individual's sequencing data is given as input to these variant callers. These results raise question about reproducibility of GATK and may account for both false positives and false negatives(?).
\end{abstract}

\keywords{genome analysis, reproducibility, repeats}

\scriptsize
\bibliographystyle{plain}
\vspace*{-0.2cm}
\bibliography{calkan}


\end{document}
