\documentclass[10pt,a4paper]{article}
\usepackage{graphicx,amssymb}
\usepackage{authblk}
\usepackage{fullpage}

\usepackage{geometry}
\geometry{lmargin=0.3in,rmargin=0.3in,tmargin=0in,bmargin=0.1in}

%\textwidth=18cm \hoffset=-0.6cm
%\textheight=28cm \voffset=-1cm


\pagestyle{empty}

\date{}

\def\keywords#1{\begin{center}{\bf Keywords}\\{#1}\end{center}}

\def\titulo#1{\title{#1}}
\def\autores#1{\author{#1}}

\begin{document}

\titulo{On Genomic Repeats and Reproducibility}

\author[1]{Can Firtina}
\author[1]{Can Alkan}
\affil[1]{\small Dept. of Computer Engineering, Bilkent University, 06800 Ankara, Turkey}
%\renewcommand\Authands{ and }

\maketitle
\thispagestyle{empty}

% The abstract

\begin{abstract}
The advancements in high throughput sequencing (HTS) technologies have increased the demand on producing  genome sequence data for many research questions. However, analyzing and interpreting HTS data with high accuracy is still a challenge. 
Although many algorithms were developed for this purpose, a handful of computational pipelines from mapping to variant calling may be considered standard, as they are commonly used in large-scale genome projects.

Approximately half of the human genome consists of repeats, which cause ambiguity in read mapping when the read length is short. On the average, a 100-bp read generated by the Illumina platform may align to hundreds of genome locations with similar edit distance. 
The BWA~\cite{Li2009a} mapper's  approach to handle such ambiguity is randomly selecting one location, and assigning the mapping quality to zero to inform the variant calling algorithms that the alignment may not be accurate. 

In this study, we investigated whether some of the commonly used variant discovery algorithms
make use of this mapping quality information, and how they react to genomic repeats.
We aligned two whole genome shotgun (WGS) datasets with one low (5X) and one high (45X) coverage genome
sequenced as part of the 1000 Genomes Project~\cite{1000GP2012} to the human reference genome (GRCh37) {\bf twice} using BWA with the same parameters. However, we shuffled the order of reads in the second mapping experiment to make sure that the same random number is not used for the same reads.
As expected, BWA reported different map locations to repetitive regions ($\sim$2.8\% of reads). Interestingly, although BWA reported zero mapping qualities for 
most of the discrepant read mappings ($\sim$94\%), it also assigned high MAPQ values ($\geq$30) for a fraction of them ($\sim$1.75\%). 
We then generated two single nucleotide polymorphism (SNP) and indel callsets each using GATK~\cite{DePristo2011} HaplotypeCaller, GATK UnifiedGenotyper, Freebayes~\cite{Garrison2012}, Platypus~\cite{Rimmer2014}, and SAMTools~\cite{Li2009b}, and structural variation (SV) callsets using DELLY~\cite{Rausch2012} from each genome.

We observed substantial differences in the callsets generated by all of the tools we tested. GATK's HaplotypeCaller showed a discrepancy of 0.4\% to 1.1\% , where UnifiedGenotyper showed the highest number of different calls between two alignments of the same data set (up to 12.76\%). As expected, 72 to 80\% of the discrepant calls were found within common repeats. However, we also observed 165 to 4,397 SNVs 
that were called within one alignment but not the other that map to coding exons. 
Furthermore, 691 of the 4397 (15.7\%) discrepant exonic SNVs predicted by GATK UnifiedGenotyper,  did not intersect with any common repeats or segmental duplications.
Freebayes, Platypus, and SAMtools predictions were more reproducable, as $>$98.5\% of the calls were identical.
DELLY also predicted different calls: $\sim$3\% of deletion, $\sim$4\% of tandem duplication, $\sim$6\% of inversion, and $\sim$3.6\% of translocation calls were spefic to a single alignment (i.e. BAM file), and $>$91\% of
these differences intersected with common repeats. More interestingly, when we ran GATK's both HaplotypeCaller and UnifiedGenotyper on the {\bf same} BAM file twice, we observed similar differences. Other tools produced no discrepancies.

Our results raise questions about reproducibility and accuracy of several commonly used genomic variation discovery tools. 
The differences in callsets we observed in this study 
may have similar sensitivity and specificity. It is expected to
have differences between different algorithms and/or parameters, but
obtaining different results should not be due to the order of {\it independently generated} reads in the input file. 
We argue that computational predictions should not change by ``luck'', and that 
one would prefer full reproducibility. 

\end{abstract}

\keywords{genome analysis, reproducibility, repeats}

\scriptsize
\bibliographystyle{plain}
\vspace*{-0.4cm}

\bibliography{calkan}


No Publishing
\end{document}
