<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"  
  "http://www.w3.org/TR/html4/loose.dtd">  
<html > 
<head><title>On Genomic Repeats and Reproducibility</title> 
<meta http-equiv="Content-Type" content="text/html; charset=iso-8859-1"> 
<meta name="generator" content="TeX4ht (http://www.tug.org/tex4ht/)"> 
<meta name="originator" content="TeX4ht (http://www.tug.org/tex4ht/)"> 
<!-- html --> 
<meta name="src" content="abstract.tex"> 
<meta name="date" content="2015-08-30 16:44:00"> 
<link rel="stylesheet" type="text/css" href="abstract.css"> 
</head><body 
>
   <div class="maketitle">


<h2 class="titleHead">On Genomic Repeats and Reproducibility</h2>
<div class="author" ><span 
class="cmr-12x-x-120">Can Firtina</span><sup class="textsuperscript"><span 
class="cmr-10x-x-109">1</span></sup> <span 
class="cmr-12x-x-120">and Can Alkan</span><sup class="textsuperscript"><span 
class="cmr-10x-x-109">1</span></sup><br /><br />
<sup class="textsuperscript"><span 
class="cmr-10x-x-109">1</span></sup><span 
class="cmr-10x-x-109">Dept. of Computer Engineering, Bilkent University, 06800 Ankara, Turkey</span></div>
<div class="date" ></div>
   </div>
   <div 
class="abstract" 
>
<div class="center" 
>
<!--l. 36--><p class="noindent" >
<!--l. 36--><p class="noindent" ><span 
class="cmbx-10x-x-109">Abstract</span></div>
      <!--l. 37--><p class="indent" >    <span 
class="cmr-10x-x-109">The advancements in high throughput sequencing (HTS) technologies have increased the demand on</span>
      <span 
class="cmr-10x-x-109">producing genome sequence data for many research questions. However, analyzing and interpreting HTS</span>
      <span 
class="cmr-10x-x-109">data with high accuracy is still a challenge. Although many algorithms were developed for this purpose,</span>
      <span 
class="cmr-10x-x-109">a handful of computational pipelines from mapping to variant calling may be considered standard, as</span>
      <span 
class="cmr-10x-x-109">they are commonly used in large-scale genome projects.</span>
      <!--l. 40--><p class="indent" >     <span 
class="cmr-10x-x-109">Approximately half of the human genome consists of repeats, which cause ambiguity in read mapping</span>
      <span 
class="cmr-10x-x-109">when the read length is short. On the average, a 100-bp read generated by the Illumina platform may</span>
      <span 
class="cmr-10x-x-109">align to hundreds of genome locations with similar edit distance. The BWA</span><span 
class="cmr-10x-x-109">&#x00A0;</span><span class="cite"><span 
class="cmr-10x-x-109">[</span><a 
href="#XLi2009a"><span 
class="cmr-10x-x-109">4</span></a><span 
class="cmr-10x-x-109">]</span></span> <span 
class="cmr-10x-x-109">mapper&#8217;s approach to</span>
      <span 
class="cmr-10x-x-109">handle such ambiguity is randomly selecting one location, and assigning the mapping quality to zero to</span>
      <span 
class="cmr-10x-x-109">inform the variant calling algorithms that the alignment may not be accurate.</span>
      <!--l. 43--><p class="indent" >     <span 
class="cmr-10x-x-109">In this study, we investigated whether some of the commonly used variant discovery algorithms make</span>
      <span 
class="cmr-10x-x-109">use of this mapping quality information, and how they react to genomic repeats. We aligned two whole</span>
      <span 
class="cmr-10x-x-109">genome shotgun (WGS) datasets with one low (5X) and one high (45X) coverage genome sequenced as</span>
      <span 
class="cmr-10x-x-109">part of the 1000 Genomes Project</span><span 
class="cmr-10x-x-109">&#x00A0;</span><span class="cite"><span 
class="cmr-10x-x-109">[</span><a 
href="#X1000GP2012"><span 
class="cmr-10x-x-109">1</span></a><span 
class="cmr-10x-x-109">]</span></span> <span 
class="cmr-10x-x-109">to the human reference genome (GRCh37) </span><span 
class="cmbx-10x-x-109">twice </span><span 
class="cmr-10x-x-109">using BWA with</span>
      <span 
class="cmr-10x-x-109">the same parameters. However, we shuffled the order of reads in the second mapping experiment to make</span>
      <span 
class="cmr-10x-x-109">sure that the same random number is not used for the same reads. As expected, BWA reported different</span>
      <span 
class="cmr-10x-x-109">map locations to repetitive regions (</span><span 
class="cmsy-10x-x-109">~</span><span 
class="cmr-10x-x-109">2.8% of reads). Interestingly, although BWA reported zero mapping</span>
      <span 
class="cmr-10x-x-109">qualities for most of the discrepant read mappings (</span><span 
class="cmsy-10x-x-109">~</span><span 
class="cmr-10x-x-109">94%), it also assigned high MAPQ values (</span><span 
class="cmsy-10x-x-109">&#x2265;</span><span 
class="cmr-10x-x-109">30) for</span>
      <span 
class="cmr-10x-x-109">a fraction of them (</span><span 
class="cmsy-10x-x-109">~</span><span 
class="cmr-10x-x-109">1.75%). We then generated two single nucleotide polymorphism (SNP) and indel</span>
      <span 
class="cmr-10x-x-109">callsets each using GATK</span><span 
class="cmr-10x-x-109">&#x00A0;</span><span class="cite"><span 
class="cmr-10x-x-109">[</span><a 
href="#XDePristo2011"><span 
class="cmr-10x-x-109">2</span></a><span 
class="cmr-10x-x-109">]</span></span> <span 
class="cmr-10x-x-109">HaplotypeCaller, GATK UnifiedGenotyper, Freebayes</span><span 
class="cmr-10x-x-109">&#x00A0;</span><span class="cite"><span 
class="cmr-10x-x-109">[</span><a 
href="#XGarrison2012"><span 
class="cmr-10x-x-109">3</span></a><span 
class="cmr-10x-x-109">]</span></span><span 
class="cmr-10x-x-109">, Platypus</span><span 
class="cmr-10x-x-109">&#x00A0;</span><span class="cite"><span 
class="cmr-10x-x-109">[</span><a 
href="#XRimmer2014"><span 
class="cmr-10x-x-109">7</span></a><span 
class="cmr-10x-x-109">]</span></span><span 
class="cmr-10x-x-109">,</span>
      <span 
class="cmr-10x-x-109">and SAMTools</span><span 
class="cmr-10x-x-109">&#x00A0;</span><span class="cite"><span 
class="cmr-10x-x-109">[</span><a 
href="#XLi2009b"><span 
class="cmr-10x-x-109">5</span></a><span 
class="cmr-10x-x-109">]</span></span><span 
class="cmr-10x-x-109">, and structural variation (SV) callsets using DELLY</span><span 
class="cmr-10x-x-109">&#x00A0;</span><span class="cite"><span 
class="cmr-10x-x-109">[</span><a 
href="#XRausch2012"><span 
class="cmr-10x-x-109">6</span></a><span 
class="cmr-10x-x-109">]</span></span> <span 
class="cmr-10x-x-109">from each genome.</span>
      <!--l. 51--><p class="indent" >     <span 
class="cmr-10x-x-109">We observed substantial differences in the callsets generated by all of the tools we tested. GATK&#8217;s</span>
      <span 
class="cmr-10x-x-109">HaplotypeCaller showed a discrepancy of 0.4% to 1.1% , where UnifiedGenotyper showed the higest</span>
      <span 
class="cmr-10x-x-109">number of different calls between two alignments of the same data set (up to 12.76%). As expected, 72 to</span>
      <span 
class="cmr-10x-x-109">80% of the discrepant calls were found within common repeats. However, we also observed 165 to 4,397</span>
      <span 
class="cmr-10x-x-109">SNVs that were called within one alignment but not the other that map to coding exons. Furthermore,</span>
      <span 
class="cmr-10x-x-109">691 of the 4397 (15.7%) discrepant exonic SNVs predicted by GATK UnifiedGenotyper, did not intersect</span>
      <span 
class="cmr-10x-x-109">with any common repeats or segmental duplications. Freebayes, Platypus, and SAMtools predictions</span>
      <span 
class="cmr-10x-x-109">were more reproducable, as </span><span 
class="cmmi-10x-x-109">&#x003E;</span><span 
class="cmr-10x-x-109">98.5% of the calls were identical. DELLY also predicted different calls:</span>
      <span 
class="cmsy-10x-x-109">~</span><span 
class="cmr-10x-x-109">3% of deletion, </span><span 
class="cmsy-10x-x-109">~</span><span 
class="cmr-10x-x-109">4% of tandem duplication, </span><span 
class="cmsy-10x-x-109">~</span><span 
class="cmr-10x-x-109">6% of inversion, and </span><span 
class="cmsy-10x-x-109">~</span><span 
class="cmr-10x-x-109">3.6% of translocation calls were</span>
      <span 
class="cmr-10x-x-109">spefic to a single alignment (i.e. BAM file), and </span><span 
class="cmmi-10x-x-109">&#x003E;</span><span 
class="cmr-10x-x-109">91% of these differences intersected with common</span>
      <span 
class="cmr-10x-x-109">repeats. More interestingly, when we ran GATK&#8217;s both HaplotypeCaller and UnifiedGenotyper on the</span>
      <span 
class="cmbx-10x-x-109">same </span><span 
class="cmr-10x-x-109">BAM file twice, we observed similar differences. Other tools produced no discrepancies.</span>
      <!--l. 58--><p class="indent" >     <span 
class="cmr-10x-x-109">Our results raise questions about reproducibility and accuracy of several commonly used genomic</span>
      <span 
class="cmr-10x-x-109">variation discovery tools. The differences in callsets we observed in this study may have similar sensitivity</span>
      <span 
class="cmr-10x-x-109">and specificity. It is expected to have differences between different algorithms and/or parameters, but</span>
      <span 
class="cmr-10x-x-109">obtaining different results should not be due to the order of </span><span 
class="cmti-10x-x-109">independently generated </span><span 
class="cmr-10x-x-109">reads in the input</span>
      <span 
class="cmr-10x-x-109">file. We argue that computational predictions should not change by &#8220;luck&#8221;, and that one would prefer</span>
      <span 
class="cmr-10x-x-109">full reproducibility.</span>
</div>
<div class="center" 
>
<!--l. 68--><p class="noindent" >
<!--l. 68--><p class="noindent" ><span 
class="cmbx-12">Keywords</span><br />
genome analysis, reproducibility, repeats</div>
   <h3 class="likesectionHead"><a 
 id="x1-1000"></a><span 
class="cmr-8">References</span></h3>
<!--l. 1--><p class="noindent" >
   <div class="thebibliography">
   <p class="bibitem" ><span class="biblabel">
 <span 
class="cmr-8">[1]</span><span class="bibsp"><span 
class="cmr-8">&#x00A0;</span><span 
class="cmr-8">&#x00A0;</span><span 
class="cmr-8">&#x00A0;</span></span></span><a 
 id="X1000GP2012"></a><span 
class="cmr-8">1000 Genomes Project Consortium. An integrated map of genetic variation from 1,092 human genomes. </span><span 
class="cmti-8">Nature</span><span 
class="cmr-8">, 491(7422):56&#8211;65, Nov 2012.</span>
   </p>
   <p class="bibitem" ><span class="biblabel">
 <span 
class="cmr-8">[2]</span><span class="bibsp"><span 
class="cmr-8">&#x00A0;</span><span 
class="cmr-8">&#x00A0;</span><span 
class="cmr-8">&#x00A0;</span></span></span><a 
 id="XDePristo2011"></a><span 
class="cmr-8">Mark</span><span 
class="cmr-8">&#x00A0;A. DePristo, Eric Banks, Ryan Poplin, Kiran</span><span 
class="cmr-8">&#x00A0;V. Garimella, Jared</span><span 
class="cmr-8">&#x00A0;R. Maguire, Christopher Hartl, Anthony</span><span 
class="cmr-8">&#x00A0;A. Philippakis, Guillermo</span>
   <span 
class="cmr-8">del Angel, Manuel</span><span 
class="cmr-8">&#x00A0;A. Rivas, Matt Hanna, Aaron McKenna, Tim</span><span 
class="cmr-8">&#x00A0;J. Fennell, Andrew</span><span 
class="cmr-8">&#x00A0;M. Kernytsky, Andrey</span><span 
class="cmr-8">&#x00A0;Y. Sivachenko, Kristian Cibulskis,</span>
   <span 
class="cmr-8">Stacey</span><span 
class="cmr-8">&#x00A0;B. Gabriel, David Altshuler, and Mark</span><span 
class="cmr-8">&#x00A0;J. Daly. A framework for variation discovery and genotyping using next-generation DNA sequencing</span>
   <span 
class="cmr-8">data. </span><span 
class="cmti-8">Nat Genet</span><span 
class="cmr-8">, 43(5):491&#8211;498, May 2011.</span>
   </p>
   <p class="bibitem" ><span class="biblabel">
 <span 
class="cmr-8">[3]</span><span class="bibsp"><span 
class="cmr-8">&#x00A0;</span><span 
class="cmr-8">&#x00A0;</span><span 
class="cmr-8">&#x00A0;</span></span></span><a 
 id="XGarrison2012"></a><span 
class="cmr-8">Erik Garrison and Gabor Marth. Haplotype-based variant detection from short-read sequencing. arXiv, Jul 2012.</span>
   </p>
   <p class="bibitem" ><span class="biblabel">
 <span 
class="cmr-8">[4]</span><span class="bibsp"><span 
class="cmr-8">&#x00A0;</span><span 
class="cmr-8">&#x00A0;</span><span 
class="cmr-8">&#x00A0;</span></span></span><a 
 id="XLi2009a"></a><span 
class="cmr-8">Heng Li and Richard Durbin.  Fast and accurate short read alignment with Burrows-Wheeler transform.  </span><span 
class="cmti-8">Bioinformatics</span><span 
class="cmr-8">, 25(14):1754&#8211;1760,</span>
   <span 
class="cmr-8">Jul 2009.</span>
   </p>
   <p class="bibitem" ><span class="biblabel">
 <span 
class="cmr-8">[5]</span><span class="bibsp"><span 
class="cmr-8">&#x00A0;</span><span 
class="cmr-8">&#x00A0;</span><span 
class="cmr-8">&#x00A0;</span></span></span><a 
 id="XLi2009b"></a><span 
class="cmr-8">Heng Li, Bob Handsaker, Alec Wysoker, Tim Fennell, Jue Ruan, Nils Homer, Gabor Marth, Goncalo Abecasis, Richard Durbin, and 1000</span>
   <span 
class="cmr-8">Genome Project Data</span><span 
class="cmr-8">&#x00A0;Processing Subgroup. The sequence alignment/map format and SAMtools. </span><span 
class="cmti-8">Bioinformatics</span><span 
class="cmr-8">, 25(16):2078&#8211;2079, Aug 2009.</span>
   </p>
   <p class="bibitem" ><span class="biblabel">
 <span 
class="cmr-8">[6]</span><span class="bibsp"><span 
class="cmr-8">&#x00A0;</span><span 
class="cmr-8">&#x00A0;</span><span 
class="cmr-8">&#x00A0;</span></span></span><a 
 id="XRausch2012"></a><span 
class="cmr-8">Tobias Rausch, Thomas Zichner, Andreas Schlattl, Adrian</span><span 
class="cmr-8">&#x00A0;M. St</span><span 
class="cmr-8">ütz, Vladimir Benes, and Jan</span><span 
class="cmr-8">&#x00A0;O. Korbel. DELLY: structural variant discovery</span>
   <span 
class="cmr-8">by integrated paired-end and split-read analysis. </span><span 
class="cmti-8">Bioinformatics</span><span 
class="cmr-8">, 28(18):i333&#8211;i339, Sep 2012.</span>
   </p>
   <p class="bibitem" ><span class="biblabel">
 <span 
class="cmr-8">[7]</span><span class="bibsp"><span 
class="cmr-8">&#x00A0;</span><span 
class="cmr-8">&#x00A0;</span><span 
class="cmr-8">&#x00A0;</span></span></span><a 
 id="XRimmer2014"></a><span 
class="cmr-8">Andy Rimmer, Hang Phan, Iain Mathieson, Zamin Iqbal, Stephen R</span><span 
class="cmr-8">&#x00A0;F. Twigg, W.</span><span 
class="cmr-8">&#x00A0;G. S500</span><span 
class="cmr-8">&#x00A0;Consortium </span><span 
class="cmr-8">&#x00A0;, Andrew O</span><span 
class="cmr-8">&#x00A0;M. Wilkie, Gil McVean,</span>
   <span 
class="cmr-8">and Gerton Lunter. Integrating mapping-, assembly- and haplotype-based approaches for calling variants in clinical sequencing applications. </span><span 
class="cmti-8">Nat</span>
   <span 
class="cmti-8">Genet</span><span 
class="cmr-8">, 46(8):912&#8211;918, Aug 2014.</span>
</p>
   </div>
<!--l. 76--><p class="indent" >   <span 
class="cmr-8">&#8221;No Publishing&#8221;</span>
    
</body></html> 



