\documentclass[10pt]{article}

%\documentclass{article}
%\documentclass[12pt]{article}%
% last revision:
\def\mydate{2005-06-09 13:58:03 brd}


%%%%%%% Two column control
\newif\ifdotwocol
\dotwocoltrue   % two col 
%\dotwocolfalse   % one col

\long\def\twocol#1#2{\ifdotwocol{#1}\else{#2}\fi}
%%%%%%%

\def\mybeforeequation{\footnotesize}
%\def\mybeforeequation{\small}
%\def\mybeforeequation{}

\def\myafterequation{\renewcommand\baselinestretch{1.1}}
%\def\myafterequation{}

%%%%%%%%%%%%%%%%
%%%%%%%%%%%%%%%%

\def\citeusmark{$^{\textstyle \star}$}
\def\citeus#1#2{\cite{#1}}

\def\crow#1#2{#2}

%\usepackage{denselists}
\usepackage{fullpage}
\usepackage[dvips]{graphicx}
\usepackage{color}
\usepackage{boxedminipage}
\usepackage{amsfonts}
\usepackage{amsmath}
\usepackage{url}
%\usepackage{times}
\usepackage{subfigure}
%\usepackage{nih}		% PHS 398 Forms
%\usepackage{nihblank}		% For printing on Blank PHS 398 Forms
%\usepackage{confidential}
\usepackage{multirow}
\usepackage[utf8]{inputenc}
%Note from brd
\long\def\todo#1{{\bf{To do:}} #1}
%\long\def\todo#1{}

\long\def\squeezable#1{#1}

%\def\a5{$\alpha_{_5}$

\def\a5{5}

%\def\mycaptionsize{\normalsize}
%\def\mycaptionsize{\small}
%\def\mycaptionsize{\small}
\def\mycaptionsize{\footnotesize}
\def\mycodesize{\footnotesize}
\def\myeqnsize{\small}

\def\sheading#1{{\bf #1:}\ }
\def\sheading#1{\subsubsection{#1}}
%\def\sheading#1{\bigskip {\bf #1.}}

\def\ssheading#1{\noindent {\bf #1.}\ } 

\newtheorem{hypothesis}{Hypothesis}
\long\def\hyp#1{\begin{hypothesis} #1 \end{hypothesis}}

\def\cbk#1{[{\em #1}]}

\def\R{\mathbb{R}}
\def\midv{\mathop{\,|\,}}
\def\Fscr{\mathcal{F}}
\def\Gscr{\mathcal{G}}
\def\Sscr{\mathcal{S}}
\def\set#1{{\{#1\}}}
\def\edge{\!\rightarrow\!}
\def\dedge{\!\leftrightarrow\!}
\newcommand{\EOP}{\nolinebreak[1]~~~\hspace*{\fill} $\Box$\vspace*{\parskip}\vspace*{1ex}}
%my way of doing starred references
\newcommand{\mybibitem}[1]{\bibitem{#1} 
\label{mybiblabel:#1}}
\newcommand{\BC}{[}
\newcommand{\EC}{]}
\newcommand{\mycite}[1]{\ref{mybiblabel:#1}\nocite{#1}}
\newcommand{\starcite}[1]{\ref{mybiblabel:#1}\citeusmark\nocite{#1}}


\def\degree{$^\circ$}
\def\R{\mathbb{R}}
\def\Fscr{\mathcal{F}}
\def\set#1{{\{#1\}}}
\def\edge{\!\rightarrow\!}
\def\dedge{\!\leftrightarrow\!}

\long\def\gobble#1{}
\def\Jigsaw{{\sc Jigsaw}}
\def\ahelix{\ensuremath{\alpha}-helix}
\def\ahelices{\ensuremath{\alpha}-helices}
\def\ahelical{$\alpha$-helical}
\def\bstrand{\ensuremath{\beta}-strand}
\def\bstrands{\ensuremath{\beta}-strands}
\def\bsheet{\ensuremath{\beta}-sheet}
\def\bsheets{\ensuremath{\beta}-sheets}
\def\hone{{\ensuremath{^1}\rm{H}}}
\def\htwo{{$^{2}$H}}
\def\cthir{{\ensuremath{^{13}}\rm{C}}}
\def\nfif{{\ensuremath{^{15}}\rm{N}}}
\def\hn{{\rm{H}\ensuremah{^\mathrm{N}}}}
\def\hnone{{\textup{H}\ensuremath{^1_\mathrm{N}}}}
\def\ca{{\rm{C}\ensuremath{^\alpha}}}
\def\catwel{{\ensuremath{^{12}}\rm{C}\ensuremath{^\alpha}}}
\def\ha{{\rm{H}\ensuremath{^\alpha}}}
\def\cb{{\rm{C}\ensuremath{^\beta}}}
\def\hb{{\rm{H}\ensuremath{^\beta}}}
\def\hg{{\rm{H}\ensuremath{^\gamma}}}
\def\dnn{{\ensuremath{d_{\mathrm{NN}}}}}
\def\dan{{\ensuremath{d_{\alpha \mathrm{N}}}}}
\def\jconst{{\ensuremath{^{3}\mathrm{J}_{\mathrm{H}^{\mathrm{N}}\mathrm{H}^{\alpha}}}} }
\def\cbfb{{CBF-$\beta$}}

\newtheorem{defn}{Definition}
\newtheorem{claim}{Claim}

    \gobble{
    \psfrag{CO}[][]{\colorbox{white}{C}}
    \psfrag{OO}[][]{\colorbox{white}{O}}
    \psfrag{CA}[][]{\colorbox{white}{\ca}}
    \psfrag{HA}[][]{\colorbox{white}{\ha}}
    \psfrag{CB}[][]{\colorbox{white}{\cb}}
    \psfrag{HB}[][]{\colorbox{white}{\hb}}
    \psfrag{HN}[][]{\colorbox{white}{\hn}}
    \psfrag{N15}[][]{\colorbox{white}{\nfif}}
    \psfrag{dnn}[][]{\dnn}
    \psfrag{dan}[][]{\dan}
    \psfrag{phi}[][]{$\phi$}
    }

\newenvironment{closeenumerate}{\begin{list}{\arabic{enumi}.}{\topsep=0in\itemsep=0in\parsep=0in\usecounter{enumi}}}{\end{list}}

%\newenvironment{closeitemize}{\begin{list}{\topsep=0in\itemsep=0in\parsep=0in}}{\end{list}}

\newenvironment{closeitemize}%
               {\begin{itemize}%
                   \setlength{\itemsep}{0pt}%
                   \setlength{\parskip}{0pt}}%
               {\end{itemize}}


\def\CR{\hspace{0pt}}           % ``invisible'' space for line break



\newif\ifdbspacing
%\dbspacingtrue  % For double spacing
\dbspacingfalse  % For normal spacing

\ifdbspacing
 \doublespacing
 \newcommand{\capspacing}{\doublespace\mycaptionsize}
\else
 \newcommand{\capspacing}{\mycaptionsize}
\fi

\def\rulefigure{\smallskip\hrule}

% \def\codesize{\normalsize}
\def\codesize{\small}

% Can use macros \be, \ee, \en as shortcuts
%  for \begin{enumerate}, \end{enumerate}, \item
%  respectively.

\def\be{\begin{enumerate}}   % Begin Enumerate
\def\ee{\end{enumerate}}     % End Enumerate
\def\en{\item}               % ENtry (item)
\def\bi{\begin{itemize}}     % Begin Itemize
\def\ei{\end{itemize}}       % End Itemize
\def\bv{\begin{verbatim}}    % Begin Verbatim
\def\ev{\end{verbatim}}      % End Verbatim

\def\matlab{{\sc matlab} }
\def\amber{{\sc amber} }
\def\KS{{$K^*$}}
\def\KSM{{K^*}} % K-Star Math
\def\KSTM{{\tilde{K}^*}}  % K-Star Tilde Math (appx K*)
\def\KOP{{$K^{\dagger}_{o}$}}  % K-Star Optimal partial
\def\KOPM{{K^{\dagger}_{o}}}  % K-Star Optimal partial Math
\def\KP{{$K^{\dagger}$}}  % K-Star partial
\def\KPM{{K^{\dagger}}}  % K-Star partial Math
\def\KTPM{{\tilde{K}^{\dagger}}}  % K-Star Tilde partial Math
\def\KD{{$K_{_D}$}}
\def\KA{{$K_{_A}$}}
\def\qpM{{q_{_P}}}
\def\qlM{{q_{_L}}}
\def\qplM{{q_{_{PL}}}}
\def\qSplM{{q^*_{_{PL}}}}
\def\KSO{{$K^*_{o}$}} % K-Star Optimal
\def\KSOM{{K^*_{o}}}  % K-Star Optimal Math
\def\CBFB{{CBF-$\beta$}}   % Core binding factor beta
\def\argmin{\mathop{\mathrm{argmin}}}
\def\rhl#1{{\em \underline{RYAN}: *\{{#1}\}*}}
\def\set#1{{\left\{ #1 \right\}}}
\def\Escr{{\mathcal{E}}}
\def\Jscr{{\mathcal{J}}}
\def\Kscr{{\mathcal{K}}}
\def\th{{$^{{\mathrm{th}}}$}}

\newtheorem{proposition}{Proposition}
\newtheorem{lemma}{Lemma}


\newcommand{\junk}[1]{}

\def\thesection{\Alph{section}}

%\textheight 9.0in 

%\addtolength{\footheight}{-20pt}
%\bibliographystyle{plos}

\usepackage{wallpaper}
\ULCornerWallPaper{1}{letterhead-w-embo.pdf}
%\usepackage{fontspec}
%\setmainfont{Calibri} 

\begin{document}

\thispagestyle{empty}

\vspace*{0.7cm}

\begin{minipage}{0.49\textwidth}
\begin{flushleft}
  \noindent Tal Nawy\\
Associate Editor\\
Nature Methods\\
One New York Plaza, Suite 4500
New York NY 10004-1562
75 Varick Street, 9th Floor\\
New York NY 10013-1917
\end{flushleft}
\end{minipage}
\begin{minipage}{0.47\textwidth}
\begin{flushright}
%  December 15, 2012 
\today
\end{flushright}
\end{minipage} \\

\vspace*{0.7cm}  

Dear Dr. Nawy:

Attached, please find a copy of our manuscript entitled ``On Genomic Repeats and Reproducibility''
for consideration as an Analysis article in Nature Methods.

Inspired from our earlier paper ``Limitations of next-generation genome sequence assembly'', previously published in Nature Methods, here we analyze the robustness of genomic variation characterization 
using high throughput sequencing technologies.
In this manuscript, we describe the behavior of several popular read mapping algorithms when a read can be aligned to multiple locations, and how various SNV, indel, and SV calling algorithms 
handle such alignments. It is known, for example, that the BWA-MEM algorithm selects random locations among possible map location candidates, and reports a zero mapping quality. It is expected
for tha variation calling algorithms to filter such alignments accordingly, however, we show that this is not the case, and the order of reads in the input files affects the 
final variation call set due to random placement. More interestingly, we also found that, even when the same alignment file (i.e. BAM file) is used, the popular GATK tool (both HaplotypeCaller and UnifiedGenotyper) produce slightly different call sets, which we pinpoint to a randomization step in variant filtration.

The HTS platforms are now being considered to be used in clinical setting to enhance both diagnosis and treatment of patients. However, any ``medical test''
has to be proven to be both accurate and reproducible  to be reliably used in the clinic. However, the fast-evolving nature of HTS technologies make it difficult
to generate essentially same data using the same DNA resources. There is some level of undeterminism in HTS data production due to signal processing, GC content, DNA degradation, etc., but, 
we argue that the bioinformatic analyses should be deterministic, since computers are deterministic machines. Our analysis shows that this is not the case for some of the most popular variation characterization pipelines due to randomization in several steps. I am concerned that, if left undocumented, discordancies in call sets even when using the {\bf same} data
 may cause harm when HTS use becomes routine in the clinic.

We would like to note that we wish to submit this article for a {\it double blind} review. We have anonymized the manuscript and avoided any phrase and several self citations
 that might reveal our names. The manuscript has only two authors, and if accepted, we would like the author names to be ordered as: Can Firtina and Can Alkan.

We recommend referees in the area of genome analysis (David Haussler, Jim Mullikin, Steve O'Brien), and
bioinformatics (Lior Pachter, Vineet Bafna, Christina Boucher).
We request that researchers at the Broad Institute (Heng Li, Mark Daly, Bob Handsaker, Steve McCarroll) be excluded as reviewers due to conflict of interest.

Thank you for consideration of this manuscript.

Sincerely,

\vspace*{0.4cm}

\noindent\includegraphics[scale=0.3]{imza.png} 

\vspace*{0.4cm}
\noindent Can Alkan, Ph.D. \\

%Department of Computer Engineering, Bilkent University 
%Bilkent, Ankara 06800 

\end{document}


